src/main/java/org/eclipse/tractusx/ssi/lib/proof/IVerifier.java

```diff
@@ -36,15 +40,19 @@ public interface IVerifier {
    * model to get the public key of issuer.
    *
    * @param hashedLinkedData the hashed linked data
-   * @param verifiable the verifiable
-   * @return the boolean
-   * @throws UnsupportedSignatureTypeException the unsupported signature type exception
-   * @throws DidDocumentResolverNotRegisteredException the did document resolver not registered
-   *     exception
-   * @throws InvalidePublicKeyFormat the invalide public key format
-   * @throws NoVerificationKeyFoundExcpetion the no verification key found excpetion
+   * @param document {@link VerifiableCredential} the verifiable
+   * @return boolean if verified or not
+   * @throws UnsupportedSignatureTypeException
+   * @throws SignatureParseException
+   * @throws InvalidPublicKeyFormatException
+   * @throws SignatureGenerateFailedException
+   * @throws SignatureVerificationFailedException
+   * @throws DidParseException
+   * @throws DidDocumentResolverNotRegisteredException
+   * @throws NoVerificationKeyFoundException
    */
   public boolean verify(HashedLinkedData hashedLinkedData, Verifiable verifiable)
-      throws UnsupportedSignatureTypeException, DidDocumentResolverNotRegisteredException,
-          InvalidePublicKeyFormat, NoVerificationKeyFoundExcpetion;
+      throws SignatureParseException, DidParseException, InvalidPublicKeyFormatException,
```

Comment:
Grouping exception in package **org.eclipse.tractusx.ssi.lib.exception.proof** classes would make it easier for clients to handle exceptions (throwing 7 different exceptions is way to many IMHO)...
### Suggestion for groups / hierarchies:

- JwtParseException: Parent for for all exceptions related to parsing a proof, e.g. SignatureParseException, 
- JwtVerificationException: Parent for for all exceptions related to verifying a proof, e.g. UnsupportedVerificationMethodException, SignatureVerificationFailedException, NoVerificationKeyFoundException, etc.
- JwtValidationException: : Parent for for all exceptions related to validating a proof, e.g. JwtAudienceCheckFailedException, JwtExpiredException